### PR TITLE
Ofir/tfl_1201

### PR DESCRIPTION
**Pull request by OfirZinkerz**
Every question should have a "report question" button

### CODE REVIEW

**Comment by assafshp** [comment]
please merge -conflict

File: demo/znkExercise/demo.js [hunk 1]
```diff
@@ -121,11 +122,6 @@
             };
 
             setExercise('practice', '165');
-        })
-        .run(function ($rootScope, $translate) {
```
Comment (assafshp) [inline demo/znkExercise/demo.js]: why did you remove it ?

File: demo/znkQuestionReport/demo.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        ])
+        .controller('Main', function () {
+            'ngInject';
+            var vm = this;
+            vm.questionData = {
+                questionQUID: '123456789',
+                exerciseID: '987654321',
+                exerciseTypeID: '3',
+                questionID: '65498'
+            }
```
Comment (assafshp) [inline demo/znkQuestionReport/demo.js]: I think it will be better to use more real data - the numbers here looks too weird

File: demo/znkQuestionReport/index.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <!-- bower:css -->
+    <link rel="stylesheet" href="Swiper/dist/css/swiper.css" />
+    <link rel="stylesheet" href="angular-material/angular-material.css" />
+    <!-- endbower-->
+
+    <link rel="stylesheet" href="main.css">
+
+</head>
+<body ng-app="demo" layout="column"  ng-controller="Main as vm">
+<znk-question-report report-data="vm.questionData" style="margin: 5em;"></znk-question-report>
```
Comment (assafshp) [inline demo/znkQuestionReport/index.html]: we usually don't use inline style
maybe it is ok here for the demo....but generally not...

File: dist/calls/main.css [hunk 4]
```diff
@@ -387,7 +387,7 @@ md-dialog.baseCallsModal {
         md-dialog.baseCallsModal md-dialog-content .btn-container button.primary {
           border: none;
           font-size: 14px;
-          color: #fff;
+          color: #ffffff;
```
Comment (assafshp) [inline dist/calls/main.css]: is it your change?

File: dist/main.css [hunk 5]
```diff
@@ -5483,3 +5483,582 @@ $blueBG8: #75cbe8;
...
+
+*/
+/* paths*/
+/* fonts size */
+/* full background image with retina */
+/**
+ * Tooltip Styles
+ */
+/* Add this attribute to the element that needs a tooltip */
+[data-tooltip] {
```
Comment (assafshp) [inline dist/main.css]: did you add it?

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.znkQuestionReport').controller('znkReportCtrl',
+        function($log, $mdDialog, $timeout, $http, ENV, AuthService, MailSenderService, reportData) {
+            'ngInject';
+
+            var self = this;
+            var userAuth = AuthService.getAuth();
+            var MAIL_TO_SEND = 'ofir@zinkerz.com';
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: support

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            var TEMPLATE_KEY = 'reportQuestion';
+            var EMAIL_SUBJECT = 'Report Question';
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: Has to be translated

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            self.reportData = reportData;
+            self.reportData.app = ENV.firebaseAppScopeName.split('_')[0].toUpperCase();
+            self.reportData.email = userAuth.auth.email;
+            self.reportData.message = 'Hello Support,\r\n' +
+                                        'I\'ve noticed the following error in this question:\r\n';
+
+            this.sendFrom = function () {
+                if (self.reportForm.$valid) {
+                    self.startLoader = true;
+                    self.reportData.email = self.reportData.email ? self.reportData.email : userAuth.auth.email;
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: there might be cases that we will not have email (facebook users who disabled email access). Make sure to support that

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+            var userAuth = AuthService.getAuth();
+            var MAIL_TO_SEND = 'ofir@zinkerz.com';
+            var TEMPLATE_KEY = 'reportQuestion';
+            var EMAIL_SUBJECT = 'Report Question';
+
+            self.success = false;
+            self.reportData = reportData;
+            self.reportData.app = ENV.firebaseAppScopeName.split('_')[0].toUpperCase();
+            self.reportData.email = userAuth.auth.email;
+            self.reportData.message = 'Hello Support,\r\n' +
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: Has to be translated

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+                    var emailSubject = EMAIL_SUBJECT;
+                    emailSubject += ' - ' + self.reportData.app;
+
+                    var ADD_TO_MESSAGE = '<br><br>' + 'App: ' + ENV.firebaseAppScopeName + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'Question ID: ' + self.reportData.id + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'Question QUID: ' + self.reportData.quid + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'Exercise ID: ' + self.reportData.parentId + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'Exercise Type ID: ' + self.reportData.parentTypeId + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'userEmail: ' + self.reportData.email + ' | ';
+                    ADD_TO_MESSAGE += '<br>' + 'userId: ' + userAuth.auth.uid;
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: the best will be to support translation here.....

we should ask Ariel since this is an internal email

File: src/components/znkQuestionReport/controllers/znkReport.controller.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+                    MailSenderService.postMailRequest(dataToSend).then(function (res) {
+                        self.fillLoader = true;
+                        $timeout(function () {
+                            self.startLoader = self.fillLoader = false;
+                        }, 100);
+
+                        if (res.data.success) {
+                            self.success = true;
+                        }
```
Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: else...

Comment (assafshp) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: maybe $log.error

File: src/components/znkQuestionReport/scss/_tooltip.scss [hunk 12]
```diff
@@ -0,0 +1,63 @@
+/**
+ * Tooltip Styles
+ */
+
+/* Add this attribute to the element that needs a tooltip */
+[data-tooltip] {
```
Comment (assafshp) [inline src/components/znkQuestionReport/scss/_tooltip.scss]: ?

Comment (assafshp) [inline src/components/znkQuestionReport/scss/_tooltip.scss]: Isn't it is too global

File: src/components/znkQuestionReport/templates/znkReport.template.html [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+                                    fill-loader="vm.fillLoader"
+                                    show-loader="vm.startLoader"
+                                    bg-loader="'#72ab40'"
+                                    precentage="50"
+                                    font-color="'#FFFFFF'"
+                                    bg="'#87ca4d'">
+                                <span translate=".SEND"></span>
+                            </button>
+                            <!--<div class="user-details-border"></div>-->
+                            <!--<div class="user-email" ng-if="vm.userEmail" translate=".USER_EMAIL"-->
```
Comment (assafshp) [inline src/components/znkQuestionReport/templates/znkReport.template.html]: if not need - delete it

File: demo/znkExercise/demo.js (hunk 1 above)
Comment (OfirZinkerz) [inline demo/znkExercise/demo.js]: Oded remove it when he help me, probably by mistake (I'll ask him when he come)
I restore the code and check it and there was no error.

File: demo/znkQuestionReport/demo.js (hunk 2 above)
Comment (OfirZinkerz) [inline demo/znkQuestionReport/demo.js]: fixed.

File: demo/znkQuestionReport/index.html (hunk 3 above)
Comment (OfirZinkerz) [inline demo/znkQuestionReport/index.html]: I also never use inline style in the real component,
I use it because it's a demo and it's just one line.
However I've move it to style tags

File: dist/calls/main.css (hunk 4 above)
Comment (OfirZinkerz) [inline dist/calls/main.css]: yes. I've changed the color variable in _colors.scss.

File: demo/znkExercise/demo.js (hunk 1 above)
Comment (OfirZinkerz) [inline demo/znkExercise/demo.js]: Oded say it suppose to come from the translate.config.js in the index.html.
I have removed it once again and it work just fine.

File: dist/main.css (hunk 5 above)
Comment (OfirZinkerz) [inline dist/main.css]: no.
the part above the comment doesn't look familiar
the part bellow yes. it's new tooltip.scss

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 6 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: I'll change it after i'll finish to aggregate into the toefl

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 7 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: fixed.

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 9 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: fixed.

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 8 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: fixed.

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 10 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: no need.

File: src/components/znkQuestionReport/controllers/znkReport.controller.js (hunk 11 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/controllers/znkReport.controller.js]: $log.error('Error sending mail');

File: src/components/znkQuestionReport/scss/_tooltip.scss (hunk 12 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/scss/_tooltip.scss]: I don't use it anymore so I delete the file and the import

File: src/components/znkQuestionReport/templates/znkReport.template.html (hunk 13 above)
Comment (OfirZinkerz) [inline src/components/znkQuestionReport/templates/znkReport.template.html]: done
